pkg/abcupdater/version.go

```diff
@@ -238,6 +234,33 @@ func CheckAppVersion(ctx context.Context, params *CheckVersionParams) (string, e
 	return "", nil
 }
 
+// asyncFunctionCall handles the async part of CheckAppVersion, but accepts
+// a function other than CheckAppVersionSync for testing.
+func asyncFunctionCall(ctx context.Context, funcToCall func() (string, error), outFunc func(string)) func() {
+	updatesCh := make(chan string, 1)
+
+	go func() {
+		defer close(updatesCh)
+		message, err := funcToCall()
+		if err != nil {
+			logging.FromContext(ctx).WarnContext(ctx, "failed to check for new versions",
+				"error", err)
+		}
+		updatesCh <- message
```

Comment:
It seems a little strange to send this message in the error case. If funcToCall() returns error, then it seems cleanest to make no assumption about the value of message and just send nothing. This combines well within another cleanup, which is to check for a closed channel in the select block, like:

```
func asyncFunctionCall(ctx context.Context, funcToCall func() (string, error), outFunc func(string)) func() {
	updatesCh := make(chan string, 1)

	go func() {
		defer close(updatesCh)
		message, err := funcToCall()
		if err != nil {
			logging.FromContext(ctx).WarnContext(ctx, "failed to check for new versions",
				"error", err)
			return // ********** HERE
		}
		updatesCh <- message
}()

	return func() {
		select {
		case <-ctx.Done():
			// Context was cancelled
		case msg, ok := <-updatesCh:  // ********** HERE
			if ok {
				outFunc(msg)
			}
		}
	}
}

Reply:
Thanks, added